arangod/Aql/SortExecutor.cpp

```diff
@@ -135,146 +134,109 @@ SortExecutor::~SortExecutor() {
   _infos.getResourceMonitor().decreaseMemoryUsage(_memoryUsageForRowIndexes);
 }
 
-void SortExecutor::initializeInputMatrix(AqlItemBlockInputMatrix& inputMatrix) {
-  TRI_ASSERT(_input == nullptr);
-  ExecutorState state;
+void SortExecutor::consumeInput(AqlItemBlockInputRange& inputRange,
+                                ExecutorState& state) {
+  size_t numDataRows = inputRange.countDataRows();
+  size_t memoryUsageForRowIndexes =
+      numDataRows * sizeof(AqlItemMatrix::RowIndex);
 
-  // We need to get data
-  std::tie(state, _input) = inputMatrix.getMatrix();
+  _rowIndexes.reserve(numDataRows);
```

Comment:
This `reserve` call is not correct, because it will try to increase the capacity to `numDataRows`. But what if the vector already contains data?
The correct call should be `_rowIndexes.reserve(_rowIndexes.size() + numDataRows);`

Reply:
Addressed